ambassador/ambassador/ir/irtlscontext.py

```diff
@@ -140,11 +140,12 @@ def resolve(self) -> bool:
             is_valid = True
 
         # If we don't have secret info, it's worth logging.
-        if not self.secret_info:
+        if not self.secret_info and not self.hosts is None:
```

Comment:
You can probably just do

```
if self.hosts and not self.secret_info:
```

because for most things in Python, `if foo` will be `True` only if `if foo is not None` is `True`.